src/docs/swift-code-generation.md

```diff
@@ -0,0 +1,70 @@
+# Swift code generation
+
+Certain parts of the Swift ABI are poorly documented and too complex to project directly to .NET, such as metadata, async contexts, and dynamic dispatch thunks. In these cases, Swift wrappers offer the flexibility to encapsulate these poorly documented ABI components within Swift code, enabling interop through the stable ABI.
+
+The main tradeoff is between added flexibility for supporting more unsable ABI scenarios and the increased cost of shipping and trimming. To minimize complexity, we propose generating Swift wrappers only when absolutely necessary, keeping them as thin as possible.
```

Comment:
```suggestion
The main tradeoff is between more maintainable C#/Swift interop boundary and the increased cost of shipping and trimming. To minimize complexity, we propose generating Swift wrappers only when absolutely necessary, keeping them as thin as possible.
```
I do not think that this gives your flexibility. The solution without Swift wrappers would be as much as flexible, but it would be much more complicated and harder to maintain.